### ISSUE TITLE
Define CSS variables when `pydata-sphinx-theme` is not used

### ISSUE THREAD

**Issue by glemaitre** [sphinx-gallery/sphinx-gallery#1221]
Currently in sphinx-gallery as set of CSS variables (e.g. `sg-background-color`) are defined to handle dark/light modes based on a CSS selector defined by `pydata-sphinx-theme` (i.e. `data-theme=light/dark`).

It makes that `sphinx-gallery` is coupled with `pydata-sphinx-theme` and the CSS variable will exist only if the `<html>` tag define the `data-theme` selector as show here:

https://github.com/scikit-learn/scikit-learn/pull/27659/files#diff-7dc66b1264ca920ea9fd1be8da9eb0d0c1250435d13b83cd69dd14c74e9de59bR12-R14

Somehow, these CSS variables are useful on our hand because it allows to detect the context where we draw the diagram (notebook, IDE, HTML page).

I was wondering if in the context of not using `pydata-sphinx-theme`, these CSS variables could still exist with a default light mode without the need to create the CSS selector from `pydata-sphinx-theme`, that could be quite tricky to debug when something goes wrong.

NB: I am far to be fluent with the HTML/CSS principles and stacks and I am open to any solution if it appears that we should do differently on sckit-learn side.

**Comment by drammock** [comment]
just a note to say that this isn't especially unique to `pydata-sphinx-theme`. For example bootstrap [uses the same approach](https://getbootstrap.com/docs/5.3/customize/color-modes/) to dynamically changing color theme, and calls it `data-bs-theme=light/dark`, and there are several other examples out there ([one](https://blog.devgenius.io/how-to-create-a-dark-light-theme-switch-using-html-css-and-javascript-d8e466dcd979) [two](https://www.section.io/engineering-education/adding-dark-theme-to-your-site/) [three](https://dev.to/ananyaneogi/create-a-dark-light-mode-switch-with-css-variables-34l8) [four](https://whitep4nth3r.com/blog/best-light-dark-mode-theme-toggle-javascript/)) that use the same `data-theme` variable name that pydata-sphinx-theme does.

**Comment by lucyleeow** [comment]
@larsoner thoughts on this? CSS stuff goes over my head.

**Comment by lucyleeow** [comment]
Also I wonder if we (scikit-learn) should put efforts into converting to pydata theme, but I am not sure who is well versed in html and has the interest...

**Comment by GaelVaroquaux** [comment]
> Also I wonder if we (scikit-learn) should put efforts into converting to pydata theme,

We (scikit-learn) should. Chiara Marmo started this work. It is a substantial effort and we should probably create a feature branch from her work and iterate on it.

**Comment by glemaitre** [comment]
> Also I wonder if we (scikit-learn) should put efforts into converting to pydata theme,

Yes we should for sure. +1 with @GaelVaroquaux thoughts.

> but I am not sure who is well versed in html and has the interest...

I would be interested to have the thing move forward. I assume that the theme should reduce the tricks that we are doing (or we should contribute upstream). I am also far to be know what I am doing with HTML/CSS but we can have a first good-looking enough version and one day, we might get feedbacks/help from UX + frontend developers.

**Comment by lucyleeow** [comment]
Yes, and maybe it is enough to have someone good with HTML/CSS to review the work.
